src/wscript

```diff
@@ -119,6 +126,67 @@ def build_tools(bld):
             use = ['gaia2'] + bld.env['USELIB'])
 
     # Cyclops server
+    if sys.platform.startswith('linux') and gaia_qt5:
+        if bld.env['WITH_CYCLOPS']:
+            # cyclops "normal" server
+            bld(
+                features = 'qt5 qt5network cxx cxxprogram cyclops',
+                source = '''
+                tools/cyclops_server/cyclopsserver.cpp
+                tools/cyclops_server/cyclopsproxy.cpp
+                tools/cyclops_server/cyclops.cpp
+                tools/cyclops_server/yamlrpcserver.cpp
+                ''',
+                target = 'tools/cyclops',
+                includes = [ '.', 'metrics', '3rdparty' ],
+                use = ['gaia2'] + bld.env['USELIB'])
+
+        # cyclops "master" server
+            bld(
+                features = 'qt5 qt5network cxx cxxprogram cyclops',
+                source = '''
+                tools/cyclops_server/cyclopsmasterserver.cpp
+                tools/cyclops_server/cyclopsproxy.cpp
+                tools/cyclops_server/cyclopsmaster.cpp
+                tools/cyclops_server/cyclops.cpp
+                tools/cyclops_server/yamlrpcserver.cpp
+                ''',
+                target = 'tools/cyclopsmaster',
+                includes = [ '.', 'metrics', '3rdparty', 'tools/cyclops_server/' ],
+                use = ['gaia2'] + bld.env['USELIB'])
+        # cyclops init scripts
+        bld.install_files('/etc/cyclops/', 'tools/cyclops_server/master.yaml')
+        bld.install_files('/etc/init.d', bld.path.ant_glob('tools/cyclops_server/initscripts/*'), chmod='0755')
+        
+    else:
+        if bld.env['WITH_CYCLOPS']:
+            # cyclops "normal" server
+            bld(
+                features = 'qt4 cxx cxxprogram cyclops',
+                source = '''
+                tools/cyclops_server/cyclopsserver.cpp
+                tools/cyclops_server/cyclopsproxy.cpp
+                tools/cyclops_server/cyclops.cpp
+                tools/cyclops_server/yamlrpcserver.cpp
+                ''',
+                target = 'tools/cyclops',
+                includes = [ '.', 'metrics', '3rdparty' ],
+                use = ['gaia2'] + bld.env['USELIB'])
+
+        # cyclops "master" server
+            bld(
+                features = 'qt4 cxx cxxprogram cyclops',
+                source = '''
+                tools/cyclops_server/cyclopsmasterserver.cpp
+                tools/cyclops_server/cyclopsproxy.cpp
+                tools/cyclops_server/cyclopsmaster.cpp
+                tools/cyclops_server/cyclops.cpp
+                tools/cyclops_server/yamlrpcserver.cpp
+                ''',
+                target = 'tools/cyclopsmaster',
+                includes = [ '.', 'metrics', '3rdparty', 'tools/cyclops_server/' ],
+                use = ['gaia2'] + bld.env['USELIB'])
+        # cyclops init scripts
     if bld.env['WITH_CYCLOPS']:
```

Comment:
Make sure that build commands aren't run twice in the case of linux.

Reply:
`.waf` runs some `cyclops` compiles twice and there is no easy way to avoid that.